lib/spack/spack/compilers/msvc.py

```diff
@@ -169,8 +163,7 @@ def __init__(self, *args, **kwargs):
         paths = args[3]
         # This positional argument "cspec" is also parsed and handled by the base class
         # constructor
-        cspec = args[0]
-        new_pth = [pth if pth else get_valid_fortran_pth(cspec.version) for pth in paths]
+        new_pth = [pth if pth else get_valid_fortran_pth() for pth in paths]
```

Comment:
I think I understand what you mean (just call this method once and populate both fortran `paths` entries if necessary) but the use of "it" is making me less than confident. Am I roughly on the right track there?

Eventually this will return different IFX versions per cl version based on VS integration and version compatibility by that'll have to wait on the try compiler work.

Reply:
> I think I understand what you mean (just call this method once and populate both fortran paths entries if necessary)

yes, or I mean, you can call it more than once, but in this list comprehension it looks like it can occur an arbitrary number of times, so like

```
default = get_valid_fortran_pth()
new_pth = [pth or default for pth in paths]
```

Reply:
Ok, we're on the same page, thanks!